irohad/consensus/yac/storage/impl/yac_proposal_storage.cpp

```diff
@@ -105,25 +102,32 @@ namespace iroha {
       // --------| private api |--------
 
       bool YacProposalStorage::shouldInsert(const VoteMessage &msg) {
-        return checkProposalHash(msg.hash.proposal_hash)
-            and checkPeerUniqueness(msg);
+        return checkProposalHash(msg.hash.proposal_hash) and checkPeerUniqueness(msg);
       }
 
       bool YacProposalStorage::checkProposalHash(ProposalHash vote_hash) {
         return vote_hash == hash_;
       }
 
+        //TODO: it is not really clear what "peer uniqueness" means. Either a better name or comments
+        // are needed here.
       bool YacProposalStorage::checkPeerUniqueness(const VoteMessage &msg) {
+            //TODO: rather than having complex return statements like this, it is usually a better design decision
+            // to encapsulate some of the logic in other functions (that are then easily testable using unit tests;
+            // the below code is extremely fragile)
         return std::all_of(block_storages_.begin(),
                            block_storages_.end(),
                            [&msg](YacBlockStorage &storage) {
+
                              if (storage.getStorageHash() != msg.hash) {
                                return true;
                              }
                              return not storage.isContains(msg);
                            });
       }
 
+        //TODO: this function is fairly complex and you should probably refactor it to be several functions that are
```

Comment:
Yes, it is possible to refactor this function with separated functions. But in this case, a part of implementation will leak to the interface. And this will be not so elegant solution. On the other hand, we may test only implementation and call methods only related to the implementation. So, we will test the implementation that also not an elegant solution, because it leads to single-use tests.